upi/openstack/control-plane.yaml

```diff
@@ -45,7 +45,7 @@
 
   - name: 'List the Server groups'
     command:
-      cmd: "openstack server group list -f json -c ID -c Name"
+      cmd: "openstack --os-compute-api-version=2.15 server group list -f json -c ID -c Name"
```

Comment:
Why 2.15? Presumably to avoid version 2.16, but that was released in 2016 and doesn't appear to affect the output of the server_groups api as far as I can tell. If we have a dependency on version 2.15, or are in some way incompatible with version 2.16 we should clearly document here what it is.

Reply:
The problem is that the library is not installed by us. This problem is caused because the user is executing ansible with an openstack-sdk that requires the library api version to be specified. 
See https://bugzilla.redhat.com/show_bug.cgi?id=1891816

Reply:
@mdbooth Actually, it's 2.15 because [the official docs for the latest version](https://docs.openstack.org/python-openstackclient/latest/cli/command-objects/server-group.html) advise so when treating with "soft" anti-affinity policies.

Reply:
That's a minimum. My comment about the 'fix' being coincidental stands because we were previously using latest, which is also >=2.15.

Reply:
I have no reason to think that bug 1891816 is related here.

If I stick to the idea that they're separate, it makes sense.

Think of this: the OP says that only the _second time_ they run it, it fails.

* `list` succeeds
* `list` fails the second time, because there's already a server group with a "*soft*" policy

In the end: the docs tell us to set a microversion. We don't set the microversion. We get a report that the thing fails, and it doesn't fail any more with a microversion.

The behaviour is not about the SDK (at least not directly), but about `python-openstackclient`.